pinot-spi/src/main/java/org/apache/pinot/spi/eventlistener/query/PinotBrokerQueryEventListenerUtils.java

```diff
@@ -0,0 +1,105 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.eventlistener.query;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.Collections;
+import java.util.Optional;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.pinot.spi.utils.CommonConstants.CONFIG_OF_BROKER_EVENT_LISTENER_CLASS_NAME;
+import static org.apache.pinot.spi.utils.CommonConstants.DEFAULT_BROKER_EVENT_LISTENER_CLASS_NAME;
+
+
+public class PinotBrokerQueryEventListenerUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PinotBrokerQueryEventListenerUtils.class);
+  private static BrokerQueryEventListener _brokerQueryEventListener = null;
+
+  private PinotBrokerQueryEventListenerUtils() {
+  }
+
+  /**
+   * Initialize the BrokerQueryEventListener and registers the eventListener
+   */
+  @VisibleForTesting
+  public synchronized static void init(PinotConfiguration eventListenerConfiguration) {
+    // Initializes BrokerQueryEventListener.
+    initializeBrokerQueryEventListener(eventListenerConfiguration);
+  }
+
+  /**
+   * Initializes PinotBrokerQueryEventListener with event-listener configurations.
+   * @param eventListenerConfiguration The subset of the configuration containing the event-listener-related keys
+   */
+  private static void initializeBrokerQueryEventListener(PinotConfiguration eventListenerConfiguration) {
+    String brokerQueryEventListenerClassName =
+        eventListenerConfiguration.getProperty(CONFIG_OF_BROKER_EVENT_LISTENER_CLASS_NAME,
+            DEFAULT_BROKER_EVENT_LISTENER_CLASS_NAME);
+    LOGGER.info("{} will be initialized as the PinotBrokerQueryEventListener", brokerQueryEventListenerClassName);
+
+    Optional<Class<?>> clazzFound;
+    try {
+      clazzFound = Optional.of(Class.forName(brokerQueryEventListenerClassName));
+    } catch (ClassNotFoundException e) {
+      throw new RuntimeException("Failed to initialize BrokerQueryEventListener. "
+          + "Please check if any pinot-event-listener related jar is actually added to the classpath.");
+    }
+
+    clazzFound.ifPresent(clazz -> {
+      try {
+        BrokerQueryEventListener brokerQueryEventListener = (BrokerQueryEventListener) clazz.newInstance();
```

Comment:
sorry missed this in previous review: should we pass in `eventListenerConfiguration` to the query event listener constructor?

That would allow custom implementations to easily configure their query event listener from the same properties file that Pinot uses.

Reply:
Added an `init` method for this.